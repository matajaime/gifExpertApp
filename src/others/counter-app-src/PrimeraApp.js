import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo = 'SubTitulo'}) => {
    let miSaludo = 'Mi Saludo';
    //console.log(props);
    if ( saludo ) {
        miSaludo = saludo;
    };
    return (
    <Fragment>
        <h1>{ miSaludo }</h1>
        <p>{ subtitulo }</p>
    </Fragment>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

export default PrimeraApp;
