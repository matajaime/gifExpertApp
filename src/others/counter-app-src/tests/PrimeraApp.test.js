//import {render} from '@testing-library/react';
import React from 'react';
import { shallow } from 'enzyme';
import PrimeraApp from '../PrimeraApp';


describe('Test Primera APP',()=>{
    /*test('Debe mostrar mensaje',()=>{
        const saludo = 'Hola CHicos';
        const {getByText} = render(<PrimeraApp saludo={saludo}/>);
        expect(getByText(saludo)).toBeInTheDocument();
    });*/
    test('Debe mostrar PrimeraApp correctamente',() => {
        const saludo = 'Hola Chicos';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe mostrar subtitulo enviado',() => {
        const saludo = 'Saludo de prueba';
        const subtitulo = "Â¡Subtitulo de Prueba!";
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);
        expect(wrapper).toMatchSnapshot();
        //buscar elemento html p (parrafo), puedes hacerlo por id '#id' o por clase '.class'
        //Si hay varios devuelve arreglo
        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe( subtitulo);
    });
});