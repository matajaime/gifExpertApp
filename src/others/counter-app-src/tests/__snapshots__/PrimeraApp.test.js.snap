// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Primera APP Debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola Chicos
  </h1>
  <p>
    SubTitulo
  </p>
</Fragment>
`;

exports[`Test Primera APP Debe mostrar subtitulo enviado 1`] = `
<Fragment>
  <h1>
    Saludo de prueba
  </h1>
  <p>
    Â¡Subtitulo de Prueba!
  </p>
</Fragment>
`;
